src/selectors/entities/channel_categories.ts

```diff
@@ -0,0 +1,250 @@
+
+            const getDisplayName = (channel: Channel): string => {
+                if (cachedNames[channel.id]) {
+                    return cachedNames[channel.id];
+                }
+
+                let displayName;
+
+                // TODO it might be easier to do this by using channel members to find the users
+                if (channel.type === General.DM_CHANNEL) {
+                    const teammateId = getUserIdFromChannelName(currentUserId, channel.name);
+                    const teammate = profiles[teammateId];
+
+                    displayName = displayUsername(teammate, teammateNameDisplay, false);
+                } else if (channel.type === General.GM_CHANNEL) {
+                    const usernames = channel.display_name.split(', ');
+
+                    const userDisplayNames = [];
+                    for (const username of usernames) {
+                        const user = Object.values(profiles).find((profile) => profile.username === username);
+
+                        if (!user) {
+                            continue;
+                        }
+
+                        if (user.id === currentUserId) {
+                            continue;
+                        }
+
+                        userDisplayNames.push(displayUsername(user, teammateNameDisplay, false));
+                    }
+
+                    displayName = userDisplayNames.sort((a, b) => a.localeCompare(b, locale, {numeric: true})).join(', ');
+                } else {
+                    displayName = channel.display_name;
+                }
+
+                cachedNames[channel.id] = displayName;
+
+                return displayName;
+            };
+
+            const sorted = [...channels];
+            sorted.sort((a, b) => getDisplayName(a).localeCompare(getDisplayName(b), locale, {numeric: true}));
+            return sorted;
+        }
+    );
+}
+
+export function makeGetChannelsForCategory() {
+    const getChannelsForAllCategories = makeGetChannelsForAllCategories();
+    const filterChannelsByFavorites = makeFilterChannelsByFavorites();
+
+    const filterChannelsByType = makeFilterChannelsByType();
+    const filterAutoclosedDMs = makeFilterAutoclosedDMs();
+    const filterManuallyClosedDMs = makeFilterManuallyClosedDMs();
+
+    const sortChannelsByName = makeSortChannelsByName();
+    const sortChannelsByNameWithDMs = makeSortChannelsByNameWithDMs();
+
+    return (state: GlobalState, category: ChannelCategory) => {
+        let channels = getChannelsForAllCategories(state, category.team_id);
+
+        channels = filterChannelsByFavorites(state, channels, category.type);
+
+        channels = filterChannelsByType(state, channels, category.type);
+
+        channels = filterAutoclosedDMs(state, channels, category.type);
+
+        channels = filterManuallyClosedDMs(state, channels);
+
+        if (channels.some((channel) => channel.type === General.DM_CHANNEL || channel.type === General.GM_CHANNEL)) {
```

Comment:
How do we do filtering per different custom categories? Or is that logic not happening yet?

Reply:
That logic isn't in yet. Right now, the "categories" in the store just act as hardcoded placeholders. Once the channel IDs for each category come from the server, then this function will do that sorting. It might actually make it a bit simpler than the current logic since the stored IDs will be mostly sorted and filtered for us.